fireplace/cards/utils.py

```diff
@@ -38,6 +40,46 @@
 AT_MAX_MANA = lambda s: MANA(s) == 10
 
 
+def weighted_card_choice(source, weights: List[int], card_sets: List[str], count: int):
+	"""
+	take a list of weights and a list of card pools and produce a random weighted
+	sample without replacement.
+	len(weights) == len(card_sets) (one weight per card set)
+	"""
+
+	# special case for single weight (generic random pick)
+	if len(card_sets) == 1:
+		return [source.controller.card(card, source=source) for card in random.sample(card_sets[0], count)]
+
+	# special case for all weights equal
+	if len(set(weights)) == 1:
```

Comment:
I didn't ask you to put this in; you said you *wanted* to put this in and I didn't say no. I agree with @smallnamespace and think this is premature optimization.